micronaut-langchain4j-core/src/main/java/io/micronaut/langchain4j/store/memory/chat/neo4j/Neo4jChatMemoryStoreFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.langchain4j.store.memory.chat.neo4j;
+
+import dev.langchain4j.community.store.memory.chat.neo4j.Neo4jChatMemoryStore;
+import io.micronaut.context.annotation.EachBean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Prototype;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.util.StringUtils;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+@Factory
+@Internal
+class Neo4jChatMemoryStoreFactory {
+    @Named("neo4j")
+    @Prototype
+    Neo4jChatMemoryStore.Builder createRedisChatMemoryStoreBuilder(Neo4jChatMemoryStoreConfiguration config) {
+        Neo4jChatMemoryStore.Builder builder = Neo4jChatMemoryStore.builder();
+        if (StringUtils.isNotEmpty(config.getMemoryLabel())) {
```

Comment:
again why duplicate all the properties if we can use `@ConfigurationBuilder`

Reply:
[`Neo4jChatMemoryStore.Builder` has a method with three parameters. I left uri, user and password and use an `@ConfigurationBuilder` for the rest](https://github.com/micronaut-projects/micronaut-langchain4j/pull/173/commits/336b7cde76223c86db359b949e1b4db0504ce07b).